src/com/content-event/event-idea.js

```diff
@@ -67,48 +65,64 @@ export default class ContentEventIdea extends Component {
 	removeIdea( id, e ) {
 		id = parseInt(id);
 
-		console.log('remove:', id );
-
+		//console.log('remove:', id );
+		this.setState({'error': null});
 		if ( id ) {
 			$ThemeIdea.Remove(this.props.node.id, id)
 			.then(r => {
 				//console.log(r.ideas);
-				this.setState({"ideas": r.ideas});
+				this.setState({'ideas': r.ideas, 'enableSubmit': canHaveMoreIdeas(r.ideas)});
 			})
 			.catch(err => {
-				this.setState({"error": err});
+				this.setState({'error': 'Error processing the request. Make sure you are still logged in.'});
 			});
 		}
 		else {
-			this.setState({"error": "Problem with length"});
+			this.setState({'error': 'Unexpected error.'});
 		}
 	}
 
-	submitIdeaForm( e ) {
-		var idea = this.state.idea.trim();
-		console.log('submit:', idea);
+	checkDuplicateIdea(idea) {
+		const {processingIdea, ideas} = this.state;
+		if (processingIdea && idea == processingIdea) {
+			return true;
+		}
+		else {
+			return Object.values(ideas).indexOf(idea) > -1;
+		}
+	}
 
-		if ( idea.length > 0 && idea.length <= 64 ) {
+	submitIdeaForm( e ) {
+		let idea = this.state.idea.trim();
+		//console.log('submit:', idea);
+		if (this.checkDuplicateIdea(idea)) {
+			this.setState({
+				'error': 'Duplicated suggestion! There\'s no point in supplying the same suggestion more than once.'
+			});
+		}
+		else if ( idea.length > 0 && idea.length <= 64 ) {
+			this.setState({'enableSubmit': canHaveMoreIdeas(this.state.ideas, true), 'processingIdea': idea, 'error': null});
 			$ThemeIdea.Add(this.props.node.id, idea)
 			.then(r => {
-				console.log('r', r);
-				this.setState({"ideas": r.ideas, "idea": r.status === 201 ? "" : idea});
+				//console.log('r', r);
+				this.setState({'ideas': r.ideas, 'idea': r.status === 201 ? '' : idea, 'enableSubmit': canHaveMoreIdeas(r.ideas), 'processingIdea': null});
 			})
 			.catch(err => {
-				this.setState({"error": err});
+				this.setState({'error': 'Error processing the request. Make sure you are still logged in.', 'processingIdea': null});
 			});
 		}
 		else {
-			this.setState({"error": "Problem with length"});
+			this.setState({'error': 'Suggestion is too ' + (idea.length == 0 ? 'short.' : 'long.')});
 		}
 	}
 
 	renderIdea( id ) {
-		var idea = this.state.ideas[id];
+		const idea = this.state.ideas[id];
```

Comment:
:+1: I just want to acknowledge that this is good. I started doing this myself (adding consts) thanks to seeing it so often in your code reviews.